spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/data/TestSparkOrcReadMetadataColumns.java

```diff
@@ -100,7 +99,7 @@ public static Object[] parameters() {
     return new Object[] {false, true};
   }
 
-  @Rule public TemporaryFolder temp = new TemporaryFolder();
+  @TempDir public java.nio.file.Path temp;
```

Comment:
```suggestion
  @TempDir private java.nio.file.Path temp;
```

Reply:
hm I think you either need to convert this test to a JUnit5-parameterized test or revert the changes here

Reply:
agreed, I'll revert it

Reply:
not reverting anymore, see [comment](https://github.com/apache/iceberg/pull/9341#issuecomment-1866220019)